addons/bus/static/tests/bus_tests.js

```diff
@@ -541,6 +541,38 @@ QUnit.module('Bus', {
             "reconnect",
         ]);
     });
+
+    QUnit.test(
+        "Fallback on simple worker when shared worker failed to initialize",
+        async function (assert) {
+            const originalSharedWorker = browser.SharedWorker;
+            const originalWorker = browser.Worker;
+            patchWithCleanup(browser, {
+                SharedWorker: function (url, options) {
+                    assert.step("shared-worker creation");
+                    const sw = new originalSharedWorker(url, options);
+                    // Simulate error during shared worker creation.
+                    setTimeout(() => sw.dispatchEvent(new Event("error")));
+                    return sw;
+                },
+                Worker: function (url, options) {
+                    assert.step("worker creation");
+                    return new originalWorker(url, options);
+                },
+            }, { pure: true });
+            patchWithCleanup(window.console, {
+                warn(message) {
+                    assert.step(message);
+                },
+            })
+            await makeTestEnv();
+            assert.verifySteps([
+                "shared-worker creation",
+                "Error while loading \"bus_service\" SharedWorker, fallback on Worker.",
+                "worker creation",
+            ]);
```

Comment:
Simulated failure is on `setTimeout` queue. At a glance it doesn't necessarily mean it'd called after `await makeTestEnv()`. I would put an `await deferred` that is resolved in this `setTimeout`.

Reply:
It won't be called after `makeTestEnv` indeed since `makeTestEnv` awaits the creation of the bus service. The idea behind this `setTimeout` is to let the time to the `Worker` creator to add its listeners so that the error is properly received.